test/langtools/tools/javac/doctree/MarkdownTest.java

```diff
@@ -0,0 +1,612 @@
+    Literal[CODE, pos:152, ...]
+    RawText[MARKDOWN, pos:163, .]
+  block tags: empty
+]
+*/
+
+    ///123.
+    ///
+    ///    public class HelloWorld {
+    ///        @dummy
+    ///        public static void main(String... args) {
+    ///            System.out.println("Hello World");
+    ///        }
+    ///    }
+    ///
+    ///456 {@code ...}.
+    void indentedHelloWorld() { }
+/*
+DocComment[DOC_COMMENT, pos:0
+  firstSentence: 1
+    RawText[MARKDOWN, pos:0, 123.]
+  body: 3
+    RawText[MARKDOWN, pos:10, public_class_HelloWorld_{|______...o_World");|________}|____}||456_]
+    Literal[CODE, pos:169, ...]
+    RawText[MARKDOWN, pos:180, .]
+  block tags: empty
+]
+*/
+
+    ///{@summary abc ``code-span {@dummy ...}`` def {@code ...} }
+    ///rest.
+    void codeSpanInInlineTag() { }
+/*
+DocComment[DOC_COMMENT, pos:0
+  firstSentence: 1
+    Summary[SUMMARY, pos:0
+      summary: 3
+        RawText[MARKDOWN, pos:10, abc_``code-span_{@dummy_...}``_def_]
+        Literal[CODE, pos:45, ...]
+        RawText[MARKDOWN, pos:56, _]
+    ]
+  body: 1
+    RawText[MARKDOWN, pos:58, |rest.]
+  block tags: empty
+]
+*/
+
+    ///{@summary abc
+    ///```code-block
+    ///  {@dummy ...}
+    ///```
+    ///def {@code ...} }
+    ///rest.
+    void codeBlockInInlineTag() { }
+/*
+DocComment[DOC_COMMENT, pos:0
+  firstSentence: 1
+    Summary[SUMMARY, pos:0
+      summary: 3
+        RawText[MARKDOWN, pos:10, abc|```code-block|__{@dummy_...}|```|def_]
+        Literal[CODE, pos:51, ...]
+        RawText[MARKDOWN, pos:62, _]
+    ]
+  body: 1
+    RawText[MARKDOWN, pos:64, |rest.]
+  block tags: empty
+]
+*/
+
+    ///abc `
+    ///def
+    void unmatchedBackTick() { }
+/*
+DocComment[DOC_COMMENT, pos:0
+  firstSentence: 1
+    RawText[MARKDOWN, pos:0, abc_`|def]
+  body: empty
+  block tags: empty
+]
+*/
+
+    ///{@summary abc `
+    ///def}
+    ///rest
+    void unmatchedBackTickInInline() { }
+/*
+DocComment[DOC_COMMENT, pos:0
+  firstSentence: 1
+    Summary[SUMMARY, pos:0
+      summary: 1
+        RawText[MARKDOWN, pos:10, abc_`|def]
+    ]
+  body: 1
+    RawText[MARKDOWN, pos:20, |rest]
+  block tags: empty
+]
+*/
+
+// While this is an important test case, it is also a negative one.
+// Note how the backticks "match" across the end of the inline tag.
+// That's unfortunate, but cannot reasonably be detected without
+// examining the contents of a code span.
+// Not surprisingly, most of the checks fail for this (bad) test case.
+//    ///
+//    ///{@summary abc `
+//    ///def}
+//    ///rest `more`
+//    ///
+//    void unmatchedBackTickInInline2() { }
+///*
+//DocComment[DOC_COMMENT, pos:0
+//  firstSentence: 1
+//    Summary[SUMMARY, pos:4
+//      summary: 1
+//        Erroneous[ERRONEOUS, pos:14, prefPos:37
+//          code: compiler.err.dc.unterminated.inline.tag
+//          body: abc_`|_def}|_rest_`more`
+//        ]
+//    ]
+//  body: empty
+//  block tags: empty
+//]
+//*/
```

Comment:
Please update the DocComment printout in that commented out test: the actual content is different. It would be nice if the test were passing at least at the moment of its initial commit.

Here's what I see locally:

```
Expect:
DocComment[DOC_COMMENT, pos:0
  firstSentence: 1
    Summary[SUMMARY, pos:4
      summary: 1
        Erroneous[ERRONEOUS, pos:14, prefPos:37
          code: compiler.err.dc.unterminated.inline.tag
          body: abc_`|_def}|_rest_`more`
        ]
    ]
  body: empty
  block tags: empty
]

Found:
DocComment[DOC_COMMENT, pos:0
  firstSentence: 1
    Summary[SUMMARY, pos:1
      summary: 1
        Erroneous[ERRONEOUS, pos:11, prefPos:32
          code: compiler.err.dc.unterminated.inline.tag
          body: abc_`|def}|rest_`more`
        ]
    ]
  body: empty
  block tags: empty
]
```